GamebookEngine/Views/Editing/Page List/PagesTableViewController.swift

```diff
@@ -13,9 +13,12 @@ protocol PagesTableViewDelegate: AnyObject {
     func deletedPage(_ page: Page)
 }
 
-class PagesTableViewController: UITableViewController {
+class PagesTableViewController: UITableViewController, UISearchBarDelegate {
+    @IBOutlet weak var pagesSearchBar: UISearchBar!
     var game: Game
     var pages: [Page] = []
+    var searching = false
+    var filteredPages: [Page] = []
```

Comment:
Creating a filtered array is how I handled showing searched content without affecting the original array in previous projects.

Reply:
I guess I can think of two reasons to possibly be against a secondary array for searching:

1.) Memory concerns. While we might not encounter this in Gamebook Engine, there could be cases (in other apps) where it's better for memory usage if we don't have two arrays. Like if the `pages` array has hundreds of megabytes of data in it, could be better to just replace the content with the filtered pages array. Basically "why keep data around in memory if we don't need it?" There could be an opposing argument, that it's better to keep the array around since people may switch in and out of searches quickly, but I think the I/O of fetching the pages from the database is pretty painless.

2.) Keeping the code concise. If we just have the one `pages` array, we can get rid of those `if searching { ... }` checks elsewhere in in this file. Basically it means all the other functions don't really need to be aware of the searching state, and if we build in other functions that replace or modify the `pages` array, those functions don't need to be "aware" of where to find those changes. Does that make sense?

Reply:
Makes sense. In that case, we should modify the pages array when the user uses the search bar. What about the case where the user clears the search bar? Would ```loadPages()``` be called again?

Reply:
Yup, I'd think so.

Reply:
Looking good! Gonna mark this one as resolved.